### PR TITLE
feat(provider/appengine): allow gcloud version to be configurable

### PR DESCRIPTION
**Pull request by danielpeach**
https://github.com/spinnaker/spinnaker/issues/2130 

Needs Halyard support.

### CODE REVIEW

File: clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/config/AppengineConfigurationProperties.groovy [hunk 1]
```diff
@@ -84,6 +85,11 @@ class AppengineConfigurationProperties {
     static String responseToString(Response response) {
       new String(((TypedByteArray) response.body).bytes)
     }
+
+    static enum GcloudVersion {
```
Comment (jtk54) [inline clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/config/AppengineConfigurationProperties.groovy]: If you manage applications that require different versions of the library, do you need multiple clouddrivers that are configured differently?

E.g. one python application `gcloud app deploy ...` and one java application `gcloud beta app deploy ...`?

Comment (danielpeach) [reply clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/config/AppengineConfigurationProperties.groovy]: Good question - it's per account, so you wouldn't need different Clouddrivers. An account maps to an App Engine application (a single service or small set of services), which is why this feels like the right granularity to me.

Comment (jtk54) [reply clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/config/AppengineConfigurationProperties.groovy]: SGTM.

**Comment by lwander** [comment]
Doesn't gcloud have a proper version? What about gcloudReleaseTrack?

**Comment by danielpeach** [comment]
Yep, that is better.

**Pull request merged**